pkg/model/run.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2019 One Concern
+ *
+ */
+
+package model
+
+import (
+	"time"
+)
+
+const (
+	Version = 1
+)
+
+type Category struct {
+	Name        string      `json:"name" yaml:"name"`               // Unique name for the category
+	Message     string      `json:"message" yaml:"message"`         // Message describing the category.
+	Contributor Contributor `json:"contributor" yaml:"contributor"` // Credentials for who created the category.
+	Version     int         `json:"version" yaml:"version"`         // Version of the run Schema.
+}
+
+type Run struct {
+	Name           string      `json:"name" yaml:"name"`                     // Name for the run
+	InstanceID     string      `json:"instanceid" yaml:"instanceid"`         // External ID used for a run. Not guaranteed to be unique.
+	RunID          string      `json:"id" yaml:"id"`                         // Internally generated Unique id for an instance of a run. Each time the same Type+Name is run it results in a new ID.
+	Timestamp      time.Time   `json:"timestamp" yaml:"timestamp"`           // Timestamp for when the run was created.
+	Contributor    Contributor `json:"contributor" yaml:"contributor"`       // Contributor who created the run.
+	RunFile        string      `json:"runfile" yaml:"runfile"`               // Raw file describing the run. Example: A yaml file.
+	Version        int         `json:"version" yaml:"version"`               // Version of the run Schema.
+	PipelineEngine string      `json:"pipelineengine" yaml:"pipelineengine"` // The entity that executes the run.
```

Comment:
Could you give an example of what a PipelineEngine might be?

Reply:
This would be argo in the initial implementation but could be a Kubernetes job as well.